src/cpp/KnowledgeItem.cpp

```
@@ -47,3 +47,34 @@ void KnowledgeItem::save_to(BaseStorageSaver &storage) {
 	storage.save_creation_time(creation_time());
 	storage.save_title(title());
 }
+
+
+std::pair<int, int> KnowledgeItem::_match(SearchCriteria<std::string, std::time_t> *criteria) {
+    double passed = 0, total = 0;

```

Why define as double and then silently convert to int?